### DISCUSSION TITLE
use local llama.cpp branch

### DISCUSSION THREAD

**Discussion by bmtwl** [abetlen/llama-cpp-python#1183] [Q&A] [answered]
Hello, I'm trying to do some testing of a modified branch of llama.cpp with llama-cpp-python, but can't figure out how to force it to use it. Is there a documented method to do this cleanly?
Thanks!

**Comment by vriesdemichael** [comment, marked as answer]
To install llama-cpp-python from source you need to init the submodule for llama.cpp so that the llama.cpp source can be found inside the cloned repo.

You can switch this submodule to your own dev branch of llama.cpp (you might need to add your forked repo as a remote to switch to it)
